### PR TITLE
feat: metabase workloads and scaling tweaks

### PR DESCRIPTION
**Pull request by jon-funk**
# Description

- Adds single replica metabase to pilot usage for reporting
- includes fix for CE-1285 for smoother scaling to prevent pod thrashing in test/prod
- re-enables a change filter trigger for deployer that was accidently left commented

> Note: the admin user is manually created at first-login, so a test environment admin will be created manually when this is merged into release/, and then again when we deploy it.

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] See comment for new close behavior to cleanup PVCs

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-compliance-enforcement-779-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-compliance-enforcement-779-frontend.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-compliance-enforcement/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-compliance-enforcement/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-compliance-enforcement-779-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-compliance-enforcement-779-frontend.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-compliance-enforcement/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-compliance-enforcement/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-compliance-enforcement-779-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-compliance-enforcement-779-frontend.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-compliance-enforcement/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-compliance-enforcement/actions/workflows/merge.yml)

### CODE REVIEW

**Comment by mishraomp** [comment]
@jon-funk @afwilcox I would say if we could reuse this, we can work together if tweaking are needed.

https://github.com/bcgov/nr-metabase

**Comment by jon-funk** [comment]
Discussed with Om - We will address the subchart issue I ran into during the Emerald move https://env-ceds.atlassian.net/browse/CE-1289

**Comment by jon-funk** [comment]
Verified new pr close behaviour https://github.com/bcgov/nr-compliance-enforcement/actions/runs/12145946943/job/33868726313

File: charts/app/templates/metabase/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      labels:
+        {{- include "metabase.selectorLabels" . | nindent 8 }}
+    spec:
+      securityContext:
+        {{- toYaml .Values.metabase.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ include "metabase.fullname" . }}
+          securityContext:
+            {{- toYaml .Values.metabase.securityContext | nindent 12 }}
+          image: "image-registry.openshift-image-registry.svc:5000/{{.Release.Namespace}}/{{ include "metabase.fullname" . }}:{{ .Values.metabase.metabaseImage.tag }}"
```
Comment (afwilcox) [inline charts/app/templates/metabase/deployment.yaml:31]: Add whitespace around {{ .Release.Namespace }} as per the sonar recommendation

File: charts/app/templates/metabase/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      {{- with .Values.metabase.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "metabase.selectorLabels" . | nindent 8 }}
+    spec:
+      securityContext:
+        {{- toYaml .Values.metabase.podSecurityContext | nindent 8 }}
+      containers:
```
Comment (afwilcox) [inline charts/app/templates/metabase/deployment.yaml:27]: As per the sonar warning, Are we able to put a request / reservation limit on this container to ensure that we have sufficient quota to spin up the container?

**Review by afwilcox — CHANGES_REQUESTED** [review]
Finally are you able to create a new branch for this work off the release/0.6.9 branch and include it in there (you should be able to cherry pick the commits the CE-1254 branch).  

This way we stop carrying around all these extra commits from previous releases that are causing some merge conflicts.

I can also do this as a final step once all the changes are made if you need me to,

File: charts/app/templates/metabase/deployment.yaml (hunk 2 above)
Comment (jon-funk) [reply charts/app/templates/metabase/deployment.yaml:27]: tested this requested change - as suspected we cannot specify ephemeral storage for pods as we do not control the nodes. So this sonar issue is effectively a wontfix.

**Comment by jon-funk** [comment]
Closing to push up new clean branch

**Pull request closed without merging**